docs/model.rst

```diff
@@ -5,21 +5,152 @@
 
-.. note:: This documentation needs to be reworked to be easier to read!
-
 .. php:class:: Model
 
+Probably the most significant class in ATK Data - Model - acts as a Parent for all your
+entity classes::
+
+   class User extends \atk4\data\Model
+
+You must define individual classes for all your business entities. Other frameworks may rely
+on XML or annotations, in ATK everything is defined inside your "Model" class through
+pure PHP (See Initialization below)
+
+Once you create instance of your model class, it can be recycled. With a single
+object you can load/unload individual records (See Single record Operations below)::
+
+   $m = new User($db);
+
+   $m->load(3);
+   $m['note'] = 'just updating';
+   $m->save();
+   $m->unload();
+
+   $m->load(8);
+   ....
+
+and even perform operations on multiple records (See Multiple record Operations below)::
+
+   $m = new User($db);
+   $m->addCondition('expired', true);
+
+   $m->deleteAll();
+
+When data is loaded from associated Persistence, it is automatically converted into
+a native PHP type (such as DateTime object) through a process called Typecasting. Various
+rules apply when you set value for model fields (Normalization) or when data is stored
+into database that does support a field type (Serialization)
+
+Furthermore, because you define Models as a class, it is very easy to introduce your own
+extensions which may include Hooks and Actions.
+
+There are many advanced topics that ATK Data covers, such as References, Joins, Aggregation,
+SQL actions, Unions, Deep Traversal and Containment.
+
+The design is also very extensible allowing you to introduce new Field types, Join strategies,
+Reference patterns, Action types.
+
+I suggest you to read the next section to make sure you fully understand the Model and its role
+in ATK Data.
+
+
+Understanding Model
+===================
+
+Please understand that Model in ATK Data is unlike models in other data frameworks. The
+Model class can be seen as a "gateway" between your code and many other features of ATK Data.
+
+For example - you may define fields and relations for the model::
+
+   $model->addField('age', ['type'=>'number']);
+   $model->hasMany('Children', Person::class);
+
+Methods `addField` and `hasMany` will ultimatelly create and link model with a corresponding
+`Field` object and `Reference` object. Those classes contain the logic, but in 95% of the use-cases,
+you will not have to dive deep into them.
+
+Model object = Data Set
+-----------------------
+From the moment when you create instance of your model class, it represents a DataSet - set of records
+that share some common traits::
+
+   $all_users = new User($db); // John, Susan, Leo, Bill, Karen
+
+Certain operations may "shrink" this set, such as adding conditions::
+
+   $male_users = $all_users->addCondition('gender', 'M');
+
+   send_email_to_users($male_users);
+
+This essentially filters your users without fetching them from the database server. Note that most
```

Comment:
As send_email_to_users would sure fetch the records from database, I find this sentence misleading together with the sample code misleading.
Either drop the database part of the sentence, or, if you wish to explicitly point out that addCondition() does not perform a request, give it an extra sentence? Like `addContition()` does not actually load any data from the database, but all upcoming database requests for $male_users will apply this condition on request.

Reply:
addressed.